test/src/streams/streamsTest.cpp

```diff
@@ -0,0 +1,68 @@
+#include <RGLFields.hpp>
+#include <gtest/gtest.h>
+#include <utils.hpp>
+#include "testKernel.hpp"
+
+using namespace testing;
+
+class StreamsTest : public RGLAutoCleanupTest {
+protected:
+};
+
+TEST_F(StreamsTest, basic_streams)
+{
+    // Prepare two non-default streams.
+    cudaStream_t graphStream;
+    cudaStream_t copyStream;
+
+    cudaStreamCreate(&graphStream);
+    cudaStreamCreate(&copyStream);
+
+    int N = 20000;
+    size_t size = N * sizeof(float);
+
+    // Allocate input vector h_A in host memory
+    float* h_A = (float*)malloc(size);
```

Comment:
Though it's just a test, variable names could be a bit more readable.

Reply:
You're using pageable memory here. In real world scenario, pinned memory will be used (`cudaHostAlloc`/`cudaMallocHost`). Please make sure it works well. Also, it would be a good idea to inspect this test case under NVIDIA Nsight Systems. You should have one installed together with CUDA Toolkit.
Example screenshot:
![image](https://user-images.githubusercontent.com/25122246/227510287-1ecba4ce-e23c-4b2a-bf61-033c77f74f53.png)